os/sys/process.c

```diff
@@ -145,7 +145,11 @@ exit_process(struct process *p, const struct process *fromprocess)
      * deallocate state associated with this process.
      */
     for(q = process_list; q != NULL; q = q->next) {
-      if(p != q) {
+      /*
+       * since p is set to PROCESS_STATE_NONE above, it will
+       * automatically be excluded from this call list
+       */
+      if(q->state == PROCESS_STATE_RUNNING) {
```

Comment:
Is that condition necessary at all? It seems that the first thing what `call_process` does is to check whether the process is running. What is weird though is that `call_process` treats `p->state` as a bitmap by doing `p->state & PROCESS_STATE_RUNNING`. I might become more readable if we used the existing function `process_is_running` everywhere.